app/src/main/java/com/portfolio/tasky/entry/network/EntryApiCall.kt

```diff
@@ -1,17 +1,20 @@
 package com.portfolio.tasky.entry.network
 
+import com.portfolio.tasky.BuildConfig
 import com.portfolio.tasky.entry.models.AuthenticationRequest
 import com.portfolio.tasky.entry.models.AuthenticationResponse
 import com.portfolio.tasky.entry.models.RegisterRequest
 import retrofit2.Response
 import retrofit2.http.Body
+import retrofit2.http.Headers
 import retrofit2.http.POST
 
 interface EntryApiCall {
-
+    @Headers("x-api-key: ${BuildConfig.api_key}")
```

Comment:
Since all of your routes require an API key it might make sense to use an interceptor for that which will add this header for all routes 👍

Reply:
I have kept this only for passing API_KEY

I am using a different interceptor for Access_Token

Reply:
The API KEY will also be used in all the requests
So you could have another Interceptor, just for the API Key